### DISCUSSION TITLE
Volar plugin for Intellisense

### DISCUSSION THREAD

**Discussion by kingyue737** [posva/unplugin-vue-router#84] [Show and tell]
Hi @posva, hi everyone, I made a volar plugin for providing Intellisense in `<route lang="json">` custom block.
https://github.com/kingyue737/volar-plugin-vue-router

![demo](https://user-images.githubusercontent.com/40021217/197701433-e2ba300b-4247-41c6-99ea-1fc0239e717c.gif)

I'm new to OSS development. Any help and suggestion would be appreciated!

**Comment by posva** [comment]
So nice!

Do you think you could merge this with https://github.com/posva/unplugin-vue-router/blob/main/volar/index.js

**Reply by kingyue737** [reply]
Thank you for your encouragement! I'm not sure what you want to achieve 🤔.
https://github.com/posva/unplugin-vue-router/blob/main/volar/index.js is a `VueLanguagePlugin` instead of a `LanguageServicePlugin`. No need to merge them together. Or you mean put them in one repo?

It seems that https://github.com/posva/unplugin-vue-router/blob/main/volar/index.js is written for `<route lang="ts">`, not `<route lang="json | yaml">`. Do we need this feature while we already have [`definePage`](https://github.com/posva/unplugin-vue-router#definepage-in-script) ?

In fact, I haven't try `unplugin-vue-router` yet. I wrote a volar plugin since there is no completion and validation in `<route>` custom block when I use `vite-plugin-pages`. Then I found `unplugin-vue-router` would be the future. I will explore this plugin later this weekend.

**Reply by posva** [reply]
> Or you mean put them in one repo?

Yes, sorry, I meant having everything in the volar folder

> not <route lang="json | yaml">. Do we need this feature while we already have [definePage](https://github.com/posva/unplugin-vue-router#definepage-in-script) ?

It will be necessary later on for some usecases that `definePage()` cannot solve
